### DISCUSSION TITLE
Advocating for a TorBox Image for a Pi 1 Model B + (Arm V6)

### DISCUSSION THREAD

**Discussion by airgap0** [radio24/TorBox#212] [Ideas]
Trying to do things with a minimal footprint leads me to install TorBox 5.2 on a Pi 1. Having been successful with earlier Versions of TorBox (up to the latest 32 Bit image Version 4.2) led me to believe that a Pi 1 has enough muscle to run the up to date TorBox. Energywise a Pi 1 outruns Pi 3 / Pi 4 for lengths. Whe it comes to usability, once the software is installed /compiled there is no great difference in performance comparing the Pis. Having no up-to-date image for a Pi 1 (32 Bit) means to do a manual install from scratch and some hours (even days) of compiling and installing. Nevertheless I got TorBox 5.2. running on a Pi 1.

So my asking is if the TorboxTeam would be so kind an bake a 32Bit image for the lesser brethren.

If there is demand I am more than willing to describe step by step how I got TorBox 5.2. running on the arm V6 Pi. And yes, there are some pitfalls in doing this.

Regards 
Sebastian

**Comment by airgap0** [comment]
I did some screenrecording showing step by step, keystroke by key keystroke what I did in setting up my torbox on a Pi 1B+. This video is a work in progress, up to now one can see what I did. Some pitfalls as well: Some superfluous blanks that garbled the instruction plus google sabotaging `wget go1.20.2.linux-armv6.tar.gz` with 404s. If Interested the [first part is here](https://drive.google.com/file/d/1bH2CP4t0x4m2Yz1YwOqlw5wVKe4jDaxS/view?usp=sharing). Feedback would be highly appreciated. Please mind, the video - even if timelapsed - is about 900Mb. And [here](https://drive.google.com/file/d/1LGnRZ97jU0sugRbR7-mMn1-1jtlK0CU-/view?usp=sharing) you will find how I got google to deliver `go1.20.2.linux-armv6.tar.gz` in the end. To be continued...
Greez
Sebastian

**Reply by radio24** [reply]
For a Raspberry Pi OS 32bit, you must use `go1.20.2.linux-armv6l.tar.gz` (with a `l` like LIMA and **not with a one** (1)). Also, see [here](https://go.dev/dl/) for all the different packages depending on the hardware and the operating system. In this case, the commands look like the following:
```
cd ~
sudo rm -rf /usr/local/go
wget https://golang.org/dl/go1.20.2.linux-armv6l.tar.gz
sudo tar -C /usr/local -xzvf go1.20.2.linux-armv6l.tar.gz
export PATH=$PATH:/usr/local/go/bin
sudo printf "\n# Added by TorBox\nexport PATH=$PATH:/usr/local/go/bin\n" | sudo tee -a .profile
```
Also, you have to be logged into the Raspberry Pi as `torbox`, not with another username, because, in this case, the export path will not be implemented in the right `.profile` file. Also, because /home/torbox doesn't exist on your system, you got errors with the following commands:
```
# Create a folder "openvpn" for *.ovpn files
sudo mkdir /home/torbox/openvpn
sudo chown -R torbox:torbox /home/torbox/
```
Using another username during the installation will lead to error messages during the installation and later on.

I will improve the manual on these points to be sure people are not landing into a dead end.

**Comment by radio24** [comment]
Thank you for your work and support!

I understand there may be some reason to install TorBox on a Raspberry Pi 1 or 2. However, I decided not to support these older Raspberry Pi versions with a dedicated image file because of the much longer building time for the image. Even preparing the 64-bit image for the newer Raspberry Pi consumes much of my time. Even worse is the testing of the image. I must redo the entire process if I find a critical error during this test. Therefore, I will not provide alternative images for older Raspberry Pi versions.

However, I will make changes to the manual install from scratch and the installation script based on your experiences and videos. I already learned something important from watching the first minutes of your video: the [Raspberry Pi Imager](https://github.com/raspberrypi/rpi-imager), which makes it easier to install and start the Raspberry Pi OS on the SD card, even without a screen attached. I completely missed that installer.

I will start to make these improvements on the manual and the script. By modifying this post, I will keep you posted on what I did.

Improvements and modifications based on the feedback from @airgap:
- [x] Recommend the Raspberry Pi Imager as an installer for the Raspberry Pi OS
- [x] Look into the installation of go
- [x] Streamline the downloading of the TorBox menu

**Reply by airgap0** [reply]
Thanks for your interest. What I can contribute is to build / compile a torbox on a Pi1 following the manual install from scratch. There are some minor obstacles that I encountered. For example I had to install go a different way because google messed with forwarding dl-addresses. In the end I could compile tor (24hrs+) the [unredacted video](https://drive.google.com/file/d/1_Q0o37Yk6jafzTmvl1WEeoqb2UWY-B5T/view?usp=sharing) is about 3 GiB - so be warned as it shows nothing really new.

**Reply by radio24** [reply]
What would be interesting for other users would be a redacted video on which they can follow your steps.

If you really want to create an alternative image, I would put it on our server. However, let me know before you make such an image, so I can share with you what kind of preparatory work I do and how I create the image.

**Comment by airgap0** [comment]
I ran into troubles with "4. Installing the TorBox Menu ". Video  [This short video](https://drive.google.com/file/d/1Nw3redDGIHMU9-11EaEElR3gJakdJ4m-/view?usp=sharing) shows my erratic efforts with the Torbox menu. I could not work out what to d/l.  "https://www.torbox.ch/data/torbox052-20230310.zip" from the big green button or the "master.zip" as per instruction? The links "download" and "from our GitHubpage" are a bit confusing. Anyway "sudo nano .profil" should read "sudo nano .profile" IMHO

I understand maintaining the images for 64bit Pi is a lot of work. So what I can offer is: In the end I will be able to get Torbox install up and running on a Pi1 with the help of the "building from scratch". Once this is done I will make the image available. It is one of the benefits of retirement to have time for fiddling with slow and stubborn Pi1.
Regards

**Reply by radio24** [reply]
Thanks for the feedback.
[torbox052-20230310.zip](https://www.torbox.ch/data/torbox052-20230310.zip) and [master.zip](https://github.com/radio24/TorBox/archive/refs/heads/master.zip) are almost the same. However, the later could have some minor bugfixes because it is generated from the TorBox GitHub repositiry. Based on your feedback, I will dump the first, static zip and change all links to the master.zip.

**Reply by radio24** [reply]
I watched your short video. All worked as expected. The `rm -r torbox` brought up an error because before giving the extracted folder the correct name (`mv TorBox-master torbox`), we have to be sure that there is not already an old `torbox` folder around. In other words, the error message is what we expect in an ideal situation.

By using `rm -r torbox` by yourself, you removed the correctly installed torbox folder, which is needed for further installation and the proper working of the TorBox. If you have been still stuck with this problem, repeat executing these commands:
```
cd ~
wget https://github.com/radio24/TorBox/archive/refs/heads/master.zip
unzip master.zip
mv TorBox-master torbox
rm -r master.zip
```

Nevertheless, I removed the `rm -r torbox` in the manual to avoid confusion. I also fixed the typo in "profil"

**Comment by airgap0** [comment]
Thanks for your continueing support. This week was quit busy familywise so I had to put torbox on the backburner. Nevertheless I ran the instructions (build from scratch) fromt start to end, everything (tor, python pip) compiled without error. Unfortunately tor does not want to start in the configuration CLIENTS > WLAN0, INTERNET > ETH0. If I try INTERNET > WLAN0, CLIENTS > WLAN1 Torbox does not want to ssh. NGINX gives me a 502 - Bad Gateway. I will have the weekend to investigate further. Could please someone point me to the right logs (/var/log/???) that I can flea for bugs?
Regards

**Reply by radio24** [reply]
Before getting to the logs, be sure that the extent of logs is set to "high" in the [configuration sub-menu](https://www.torbox.ch/?page_id=875). After a reboot, you will find the usual log files in `/var/log` (for example, `/var/log/messages`). All tor-related logs are in `/var/log/tor` (especially `/var/log/tor/notices.log`).

If tor doesn't want to start, it could be that in `torrc` is an interface configured, which is unavailable. In `torrc`, you should see something like this (with `#` the in the last line):
```
TransPort 127.0.0.1:9040
TransPort 192.168.42.1:9040
#TransPort 192.168.43.1:9040
```
`ifconfig` should show you the interfaces you want to use, for example, `wlan0` with `192.168.42.1` and `eth0` with the IP from your Internet router. There shouldn't be another network device plugged into the Raspberry Pi (another wifi USB adapter). With `wlan0` and `wlan1`, usually `wlan0` is the integrated wifi on the Raspberry Pi chip, and `wlan1` is an additional wifi USB adapter. In this case, `wlan1` should have `192.168.42.1` and `wlan0` the IP from your Internet router. In every case, your first client should have `192.168.42.10`.

If you need additional support from me regarding networking, please check first [this](https://www.torbox.ch/?page_id=112#i-have-problems-with-the-wifi-network-fromto-torboxinternet-routerclients-connected-to-the-torbox-please-fix-it-for-me) and [this](https://www.torbox.ch/?page_id=112#i-cant-get-torbox-to-work-and-need-more-detailed-help-what-should-i-do) entries in the FAQ
